### ISSUE TITLE
Update needed for Starscream from 4.0.4 to latest

### ISSUE THREAD

**Issue by Josiah-Agosto** [aws-amplify/aws-appsync-realtime-client-ios#138]
When using Socket.IO-Client for iOS, the StarscreamAdapter+Delegate files, didReceive methods enum has updated and throws an error to add the .peerClosed case. Without this enum case added it fails and can't be fixed. I see the starscream package is set to exactly 4.0.4, as of this ticket 4.0.6 is latest, this should either be set regularly or next minor etc.

**Comment by ejubber** [comment]
+1, this is blocking for us

**Comment by Josiah-Agosto** [comment]
> +1, this is blocking for us

Yeah this issue was not even allowing me to build or do anything. As a workaround, because I doubt this will get fixed rather quickly. I changed the version from 4.0.6 (broken version) back to 4.0.4. In the code, it's setup already for that, and shows 4.0.4. However the Package.swift is setup to be .nextMinorVersion from 4.0.4. So if you set that to exact, it works. however, there are a few files that reference that under the AppSyncRealTimeClient package (SPM). If you change all of them to exact, and not from eventually it will save it in cache somewhere and not change back to 4.0.6 unless you manually update packages.

**Comment by ejubber** [comment]
Got it, thanks @Josiah-Agosto !

**Comment by atierian** [comment]
Thanks for opening this issue @Josiah-Agosto. Which version of AWS AppSync Realtime Client iOS are you using?
[v3.1.1](https://github.com/aws-amplify/aws-appsync-realtime-client-ios/releases/tag/3.1.1) was released on 8/20/23 to prevent this issue by pinning to Starscream 4.0.4

**Comment by Josiah-Agosto** [comment]
Hello @atierian, I currently have my package manager set to the main branch of Amplify, which seems to have the version at 2.19.0. I noticed that Starscream which is used in this repository refers to 4.0.4 and UP, not exactly 4.0.4, so it kept updating to 4.0.6, which is the latest in Starscream. Any other things you would like me to get, let me know.

**Comment by atierian** [comment]
The `main` branch of Amplify Swift [defines the version rules](https://github.com/aws-amplify/amplify-swift/blob/main/Package.swift#L13C1-L13C104) for AWS AppSync Realtime Client iOS as `from: 3.0.0`, which should pull in 3.1.1
Xcode is likely using a local cache - you can update by selecting `File > Packages > Update to latest package versions` in Xcode.

Side note: we recommend targeting a tagged version of Amplify (e.g. `from: 2.0.0`) rather than the `main` branch. We can't make any guarantees about the stability of `main`.

**Comment by atierian** [comment]
(I'm transferring this issue to the Amplify Swift repo)
